OrbitGl/App.cpp

```diff
@@ -1523,22 +1524,36 @@ void OrbitApp::DeselectTracepoint(const TracepointInfo& tracepoint) {
 }
 
 void OrbitApp::AddFrameTrack(const FunctionInfo& function) {
-  GetMutableCaptureData().InsertFrameTrack(function);
-  data_manager_->set_user_defined_capture_data(GetCaptureData().user_defined_capture_data());
-  AddFrameTrackTimers(function);
+  data_manager_->InsertFrameTrack(function);
+
+  if (!HasCaptureData()) {
+    return;
+  }
+  // We only add a frame track to the actual capture data if the function for the frame
+  // track actually has hits in the capture data. Otherwise we can end up in inconsistent
+  // states where "empty" frame tracks exist in the capture data (which would also be
+  // serialized).
+  const FunctionStats& stats = GetCaptureData().GetFunctionStatsOrDefault(function);
+  if (stats.count() > 1) {
+    GetMutableCaptureData().InsertFrameTrack(function);
+    AddFrameTrackTimers(function);
+  }
 }
 
 void OrbitApp::RemoveFrameTrack(const FunctionInfo& function) {
-  GetMutableCaptureData().EraseFrameTrack(function);
-  data_manager_->set_user_defined_capture_data(GetCaptureData().user_defined_capture_data());
-  GCurrentTimeGraph->RemoveFrameTrack(function);
+  data_manager_->EraseFrameTrack(function);
+  if (HasCaptureData() && GetCaptureData().ContainsFrameTrack(function)) {
+    GetMutableCaptureData().EraseFrameTrack(function);
+    GCurrentTimeGraph->RemoveFrameTrack(function);
+  }
 }
 
 bool OrbitApp::HasFrameTrack(const FunctionInfo& function) const {
```

Comment:
Where is that function called from? Just want to double check, that it will return false e.g. if there was not already a capture.

Reply:
Code changed quite a bit, could you take a look again?